mantis-runtime/src/main/java/io/mantisrx/runtime/sink/ServerSentEventsSink.java

```diff
@@ -141,57 +145,81 @@ public void call(Context context, PortRequest portRequest, final Observable<T> o
-                    .maxChunkTimeMSec(getBatchInterval());
+                .name(serverName)
+                .groupRouter(Routers.roundRobinSse(serverName, encoder))
+                .port(port)
+                .metricsRegistry(context.getMetricsRegistry())
+                .maxChunkTimeMSec(maxReadTime())
+                .maxChunkSize(maxChunkSize())
+                .bufferCapacity(bufferCapacity())
+                .numQueueConsumers(numConsumerThreads())
+                .useSpscQueue(useSpsc())
+                .maxChunkTimeMSec(getBatchInterval());
             if (predicate != null) {
                 config.predicate(predicate.getPredicate());
             }
-            PushServerSse<T, Context> server = PushServers.infiniteStreamSse(config.build(), observable,
-                    requestPreprocessor, requestPostprocessor,
-                    subscribeProcessor, context, true);
-            server.start();
+            pushServerSse = PushServers.infiniteStreamSse(config.build(), observable,
+                requestPreprocessor, requestPostprocessor,
+                subscribeProcessor, context, true);
+            pushServerSse.start();
         } else {
             LOG.info("Serving legacy HTTP SSE server sink on port: " + port);
 
             int batchInterval = getBatchInterval();
-            HttpServer<ByteBuf, ServerSentEvent> server = RxNetty.newHttpServerBuilder(port, new ServerSentEventRequestHandler<>(observable, encoder,
-                    errorEncoder, predicate, requestPreprocessor, requestPostprocessor, context, batchInterval))
-                    .pipelineConfigurator(PipelineConfigurators.<ByteBuf>serveSseConfigurator())
-                    .channelOption(ChannelOption.WRITE_BUFFER_HIGH_WATER_MARK, 5 * 1024 * 1024)
-                    .channelOption(ChannelOption.WRITE_BUFFER_LOW_WATER_MARK, 1024 * 1024)
-                    .build();
-            server.start();
+            httpServer = RxNetty.newHttpServerBuilder(
+                    port,
+                    new ServerSentEventRequestHandler<>(
+                        observable,
+                        encoder,
+                        errorEncoder,
+                        predicate,
+                        requestPreprocessor,
+                        requestPostprocessor,
+                        context,
+                        batchInterval))
+                .pipelineConfigurator(PipelineConfigurators.<ByteBuf>serveSseConfigurator())
+                .channelOption(ChannelOption.WRITE_BUFFER_HIGH_WATER_MARK, 5 * 1024 * 1024)
+                .channelOption(ChannelOption.WRITE_BUFFER_LOW_WATER_MARK, 1024 * 1024)
+                .build();
+            httpServer.start();
         }
         portObservable.onNext(port);
     }
 
+    @Override
+    public void close() throws IOException {
+        if (pushServerSse != null) {
+            pushServerSse.shutdown();
+        } else if (httpServer != null) {
```

Comment:
nit: in this case IMO null check is enough to shutdown pushServerSse and httpServer, which is slightly safer if for any reason both of them got created.